3 - FightClub/Code/FightClub.cs

```diff
@@ -7,17 +8,88 @@ public static class FightClubGame
 	{
 		public static IPowerUp PurchasePowerUp()
 		{
-			throw new NotImplementedException();
-		}
+            var powerUp = new PowerUp
+            {
+                DamageBuff = 1
+            };
+            return powerUp;
+        }
+
+
 
-		public static IPotion PurchasePotion()
+
+        public static IPotion PurchasePotion()
 		{
-			throw new NotImplementedException();
-		}
+            var potion = new Potion
+            {
+                Healing = 2,
+                Type = PotionType.Basic
+            };
+            return potion;
+        }
 
 		public static IFighter HireFighter()
 		{
-			throw new NotImplementedException();
+            var fighter = new Fighter
+            {
+                Damage = 1,
+                Health = 10
+            };
+            return fighter;
 		}
-	}
+
+        class PowerUp : IPowerUp
```

Comment:
In almost all cases, classes should exist in their own file. It not only makes it easier to code review, it makes it easier to unit test and to maintain the code. When someone see's a class being used in code and went to go fine the class it would be very confusing. There are cases where this pattern can be broken, but in most cases the industry has 1 class per 1 file.